azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -289,6 +290,9 @@ def _update(self, force_update):
         # case, to ensure it fetches the new certificate.
         #
         if self._extensions_goal_state.source == GoalStateSource.FastTrack:
+            certs_uri = findtext(xml_doc, "Certificates")
+            if certs_uri is not None:
+                self._download_certificates(certs_uri)
```

Comment:
In case if we downloaded new certs then we are not updating the state of goal_state certs in self._certs.  Next line check certificates is using self._certs which will have old certs. Am I missing something?